### PR TITLE
Add Juno bindings

### PR DESCRIPTION
**Pull request by patiee**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,34 @@
-# graph-juno-ts
-Graph Juno protobuf typescript binding
+# cosmoshub-ts
```
Comment (Seb-figment) [inline README.md]: juno-ts

File: README.md [hunk 2]
```diff
@@ -1,2 +1,34 @@
...
+
+This modules contains the typescript/assemblyscript binding for cosmos protobuf
```
Comment (Seb-figment) [inline README.md]: juno protobuf

File: package.json [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    "prebuild-ts": "sed -i -e 's[import \"proofs.proto\"[import \"confio/proofs.proto\"[' ibc-go/proto/ibc/core/commitment/v1/commitment.proto",
+    "build-ts": "protoc @protoc-options.txt $(./protoc-file-list.sh)",
+    "build": "rimraf src/* && yarn build-ts",
+    "test": "asc --explicitStart --exportRuntime --runtime stub src/index.ts --lib juno-ts -b index.wasm"
+  },
+  "dependencies": {
+    "as-proto": "^0.2.3"
+  },
+  "devDependencies": {
+    "as-proto-gen": "https://github.com/figment-networks/as-proto.git#v0.3.0-alpha.2",
```
Comment (Seb-figment) [inline package.json]: please use v0.3.0-alpha.3 and regenerate file. There was an issue with gen-dependencies in alpha.2

File: .gitmodules [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+	branch = v6.0.0
+	shallow = true
+	ignore = all
+
+[submodule "wasmd"]
+	path = wasmd
+	url = https://github.com/CosmWasm/wasmd.git
+	branch = v0.27.0
+	shallow = true
+	ignore = all
```
Comment (Seb-figment) [inline .gitmodules]: new line at end of file

File: History.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+
+0.1.0 / 2022-06-09
+==================
+
+  * Support juno v6.0.0
```
Comment (pienkowb) [inline History.md]: Such a file is usually called `CHANGELOG.md` (uppercased).

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16
+      - uses: arduino/setup-protoc@v1
+      - run: git submodule update --init --force
+      - run: yarn install --frozen-lockfile
+      - run: yarn build
+      - run: yarn test
+
```
Comment (pienkowb) [inline .github/workflows/build.yml]: Empty newline

File: README.md [hunk 7]
```diff
@@ -1,2 +1,34 @@
-# graph-juno-ts
-Graph Juno protobuf typescript binding
+# juno-ts
+
+This modules contains the typescript/assemblyscript binding for Juno protobuf
```
Comment (pienkowb) [inline README.md]: > typescript/assemblyscript

I think `as-proto` generates bindings specifically for AssemblyScript.

> binding

bindings?

> Juno protobuf

I would change that to "Juno messages".

File: package.json [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+    "prebuild-ts": "sed -i -e 's[import \"proofs.proto\"[import \"confio/proofs.proto\"[' ibc-go/proto/ibc/core/commitment/v1/commitment.proto",
+    "build-ts": "protoc @protoc-options.txt $(./protoc-file-list.sh)",
+    "build": "rimraf src/* && yarn build-ts",
+    "test": "asc --explicitStart --exportRuntime --runtime stub src/index.ts --lib juno-ts -b index.wasm"
+  },
+  "dependencies": {
+    "as-proto": "^0.2.3"
+  },
+  "devDependencies": {
+    "as-proto-gen": "https://github.com/figment-networks/as-proto.git#v0.3.0-alpha.3",
```
Comment (pienkowb) [inline package.json:20]: If we're close to merging our changes upstream, we could wait and use the official package.

File: History.md (hunk 5 above)
Comment (Seb-figment) [reply History.md]: This is the name generated by `git changelog`. I'll rename to CHANGELOG.md

File: README.md (hunk 7 above)
Comment (Seb-figment) [reply README.md]: This lib can also be used as typescript.

File: package.json (hunk 8 above)
Comment (Seb-figment) [reply package.json:20]: ok. I think it may happen this week.

File: README.md (hunk 7 above)
Comment (pienkowb) [reply README.md]: Alright, let's keep both but capitalize the names (AssemblyScript/TypeScript) and add a full stop at the end of the sentence.

File: package.json (hunk 8 above)
Comment (pienkowb) [reply package.json:20]: Ok, let's wait then.

**Review by pienkowb — APPROVED** [review]
Everything looks good 👍

@Seb-figment If we merge this now, please remember to update the `as-proto` dependency to the official package once we merge our changes upstream :)

**Pull request merged**